File: server/php/src/Application/Domain/Payments/PaymentDetails/PaymentProduct771SpecificOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class PaymentProduct771SpecificOutput
{
    public ?string $mandateReference = null;

    public function __construct(?string $mandateReference = null)
    {
        $this->mandateReference = $mandateReference;
    }

    public function toArray(): array
    {
        return [
            'mandateReference' => $this->mandateReference,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/payment_product771_specific_output.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class PaymentProduct771SpecificOutput
          attr_accessor :mandate_reference

          def initialize(mandate_reference: nil)
            @mandate_reference = mandate_reference
          end

          def to_h
            {
              'mandateReference' => @mandate_reference
            }
          end
        end
      end
    end
  end
end